vm-monitor/src/test/java/com/epam/pipeline/vmmonitor/service/impl/GCPMonitorServiceTest.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.vmmonitor.service.impl;
+
+import com.epam.pipeline.entity.region.CloudProvider;
+import com.epam.pipeline.entity.region.GCPRegion;
+import com.epam.pipeline.vmmonitor.model.vm.VirtualMachine;
+import com.epam.pipeline.vmmonitor.service.VMMonitorService;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Instance;
+import com.google.api.services.compute.model.InstanceList;
+import com.google.api.services.compute.model.NetworkInterface;
+
+import java.util.*;
+import java.math.BigInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+@SuppressWarnings("PMD.AvoidUsingHardCodedIP")
+public class GCPMonitorServiceTest {
+    private static final String CREDENTIALS_PATH = ""; // path to the Google credentials
+    private static final String PROJECT_NAME = ""; // Project ID (for example: "resounding-rune-122345")
+    private static final String REGION_CODE = ""; // the name of the zone (for example: "us-central1-a")
+    private static final String TAG = "monitored=true"; // tag for VM instances to be searched
+    private static final String TEST_INSTANCE_NAME1 = "testInstance1";
+    private static final String TEST_INSTANCE_NAME2 = "testInstance2";
+    private static final String TEST_INSTANCE_ID1 = "1";
+    private static final String TEST_INSTANCE_ID2 = "2";
+    private static final String TEST_INSTANCE_INTERNAL_IP1 = "10.128.0.1";
+    private static final String TEST_INSTANCE_INTERNAL_IP2 = "10.128.0.2";
+
+    @Spy
+    private final GCPMonitorService service = new GCPMonitorService(TAG);
+    private final GCPRegion testGcpRegion = new GCPRegion();
+
+    @Before
+    public void initializeTestGcpRegion() {
+        testGcpRegion.setAuthFile(CREDENTIALS_PATH);
+        testGcpRegion.setProject(PROJECT_NAME);
+        testGcpRegion.setRegionCode(REGION_CODE);
+    }
+
+    @Test
+    public void getProvider() {
+        final VMMonitorService service = new GCPMonitorService(TAG);
+        Assertions.assertEquals(CloudProvider.GCP, service.provider());
+    }
+
+    @Test
+    public void emptyProjectName() {
+        final GCPRegion gcpRegion = new GCPRegion();
+        gcpRegion.setAuthFile(CREDENTIALS_PATH);
+        gcpRegion.setRegionCode(REGION_CODE);
+        Assertions.assertThrows(IllegalArgumentException.class,
```

Comment:
It is common practice to static import `Assertions` class

Reply:
Done [4e8957b](https://github.com/epam/cloud-pipeline/pull/403/commits/4e8957b9096e5d0eed8d990a9ecc620c039a86b7)